### ISSUE TITLE
Emoticons : "All" section does not display all available emoticons

### ISSUE THREAD

**Issue by OlivierJaquemet** [tinymce/tinymce#6822]
**What is the current behavior? Describe the bug**

In the emoticons chooser, when the "All" section is selected, the emojis displayed in the right panel is not the sum of all emojis available in others sections.
The [emoticons documentation](https://www.tiny.cloud/docs/plugins/opensource/emoticons/) does not indicates such behavior

![2021-05-12_14-17](https://user-images.githubusercontent.com/618861/117973818-21fc1180-b32d-11eb-989f-196d9ebdfa0b.gif)

**Please provide the steps to reproduce and if possible a minimal demo of the problem via [fiddle.tiny.cloud](https://fiddle.tiny.cloud/) or similar.**

Can be reproduced by using example in emoticons documentation.
https://www.tiny.cloud/docs/plugins/opensource/emoticons/

**What is the expected behavior?**
All emoticons should be displayed and scrollable when selecting "All"

**Which versions of TinyMCE, and which browser / OS are affected by this issue? Did this work in previous versions of TinyMCE?**
TinyMCE 5.8
Plugin : emoticons

**Comment by OlivierJaquemet** [comment]
This behavior can still be observed with the latest TinyMCE version. 
Check the behavior in the [live example of the emoticons plugins documentation](https://www.tiny.cloud/docs/tinymce/latest/emoticons/)

The "all" section does not really contains all emojis available in sub section.

**Comment by TheSpyder** [comment]
We limit the results to 300 due to performance issues in our dialog framework.
https://github.com/tinymce/tinymce/blob/236891ffcdf2fd1b698d020e67c31838fb110019/modules/tinymce/src/plugins/emoticons/main/ts/ui/Dialog.ts#L21

We looked at ways to optimise this but there wasn't a clear winner, and we haven't circled back to it.

**Comment by OlivierJaquemet** [comment]
Thank you @TheSpyder for this technical information.

If there is a technical limit that cannot be overcome (for now), then the UI should probably be adapted : 
- a "load more" action 
- a text indication of the limit, "maximum number of emoji displayed, use individual sections to browse other emoji"
- ... ?

But the current limit cannot be understood by users which expect "All" to really contains everything.

**Comment by Zorandul** [comment]
I agree with Olivier... Our users using this feature complained about the same thing... They expect that "All" has all the emoticons and are suprised, when they find out its not the case...

**Comment by 0x9090** [comment]
Bumping this issue, as it is unintuitive behavior for our users as well

**Comment by OlivierJaquemet** [comment]
Bug is still relevant. Keep open little bot.

**Comment by MitchC1999** [comment]
Hi, thank you for confirming that this issue is important to you. We have logged a ticket internally to investigate.

**Comment by MitchC1999** [comment]
Hi @OlivierJaquemet, a fix for this issue was released in version 8.7. [Here is the entry in the release notes](https://www.tiny.cloud/docs/tinymce/latest/8.7.0-release-notes/#removed-the-300-item-limit-on-the-emoticons-dialog-all-tab).

**Comment by OlivierJaquemet** [comment]
Thank you, I just tested the last version on the demo page and confirm the fix.
